plugins/moondream/vision_agents/plugins/moondream/vlm/moondream_local_vlm.py

```
@@ -0,0 +1,375 @@
+
+                if isinstance(result, dict) and "answer" in result:
+                    stream = result["answer"]
+                else:
+                    stream = result
+
+                answer = await asyncio.to_thread(self._consume_stream, stream)
+
+                if not answer:
+                    logger.warning("Moondream query returned empty answer")
+                    self._is_processing = False
+                    return None
+
+                self.events.send(LLMResponseChunkEvent(delta=answer))
+                self.events.send(LLMResponseCompletedEvent(text=answer))
+                logger.info(f"Moondream VQA response: {answer}")
+                self._is_processing = False
+                return LLMResponseEvent(original=answer, text=answer)
+
+            elif self.mode == "caption":
+                self._is_processing = True
+                result = await asyncio.to_thread(self.model.caption, image, length="normal", stream=True)
+
+                if isinstance(result, dict) and "caption" in result:
+                    stream = result["caption"]
+                else:
+                    stream = result
+
+                caption = await asyncio.to_thread(self._consume_stream, stream)
+
+                if not caption:
+                    logger.warning("Moondream caption returned empty result")
+                    self._is_processing = False
+                    return None
+
+                self.events.send(LLMResponseChunkEvent(delta=caption))
+                self.events.send(LLMResponseCompletedEvent(text=caption))
+                logger.info(f"Moondream caption: {caption}")
+                self._is_processing = False
+                return LLMResponseEvent(original=caption, text=caption)
+            else:
+                logger.error(f"Unknown mode: {self.mode}")
+                self._is_processing = False
+                return None
```

This should be validated at `__init__` and fail with ValueError instead of just logging it.